htmresearch/frameworks/sp_paper/sp_metrics.py

```diff
@@ -755,4 +779,142 @@ def calculateInputSpaceCoverage(sp):
     sp.getConnectedSynapses(columnIndex, connectedSynapses)
     inputSpaceCoverage += connectedSynapses
   inputSpaceCoverage = np.reshape(inputSpaceCoverage, sp.getInputDimensions())
-  return inputSpaceCoverage
\ No newline at end of file
+  return inputSpaceCoverage
+
+
+def reconstructionError(sp, inputVectors, activeColumnVectors, threshold=0.):
+  """
+  Computes a reconstruction error. The reconstuction $r(x)$ of an input vector $x$
+  is given by the sum of the active column's connected synapses vector of 
+  the SDR representation $sdr(x)$ of $x$ normalized by $1/numActiveColumns$. 
+  The error is the normalized sum over the "hamming distance" (i.e. distance 
+  induced by L1 norm) of $x$ and its reconstruction $r(x)$, i.e. (mathy stuff in LaTex)
+  \[
```

Comment:
The threshold parameter is not explained in here. Did you use a non-zero threshold in the experiment?